src/mongo/tree/MongoCollectionTreeItem.ts

```diff
@@ -171,84 +167,75 @@ export class MongoCollectionTreeItem implements IAzureParentTreeItem {
 		}
 	}
 
-	//tslint:disable:no-any
-	private async findOne(args?: any[]): Promise<string> {
-		let result;
-		if (!args || args.length === 0) {
-			result = await this.collection.findOne({});
-		} else if (args.length === 1) {
-			result = await this.collection.findOne(args[0]);
-		} else if (args.length === 2) {
-			result = await this.collection.findOne(args[0], { fields: args[1] });
-		} else {
-			return Promise.reject(new Error("Too many arguments passed to findOne."));
-		}
+	private async findOne(query?: Object, fieldsOption?: Object): Promise<string> {
+		const result = await this.collection.findOne(query || {}, { fields: fieldsOption });
 		// findOne is the only command in this file whose output requires EJSON support.
 		// Hence that's the only function which uses EJSON.stringify rather than this.stringify.
 		return EJSON.stringify(result, null, '\t');
 	}
 
 	private insert(document: Object): Thenable<string> {
+		if (!document) {
+			throw new Error("The insert command requires at least one argument");
+		}
 		return this.collection.insert(document)
 			.then(({ insertedCount, insertedId, result }) => {
 				return this.stringify({ insertedCount, insertedId, result });
 			});
 	}
 
-	private insertOne(document: Object): Thenable<string> {
-		return this.collection.insertOne(document)
+	// tslint:disable-next-line:no-any
+	private insertOne(document: Object, options?: any): Thenable<string> {
+		return this.collection.insertOne(document, { w: options && options.writeConcern })
 			.then(({ insertedCount, insertedId, result }) => {
 				return this.stringify({ insertedCount, insertedId, result });
 			});
 	}
 
 	//tslint:disable:no-any
-	private insertMany(args: any[]): Thenable<string> {
-		// documents = args[0], collectionWriteOptions from args[1]
+	private insertMany(documents: any[], options?: any): Thenable<string> {
 		let insertManyOptions: CollectionInsertManyOptions = {};
 		const docsLink: string = "Please see mongo shell documentation. https://docs.mongodb.com/manual/reference/method/db.collection.insertMany/#db.collection.insertMany.";
-		if (!args || args.length === 0) {
+		if (!documents) { // this code should not be hit
 			return Promise.reject(new Error("Too few arguments passed to insertMany. " + docsLink));
```

Comment:
Using an assert here throws an error inside a promise. 
Is there a way to return a rejected promise in case an assertion fails?